Mfc example

bsdjhb opened the pull request:
I might try to find a non-conflicting example merge to add as well before the conflicting one.

bsdjhb committed:
Restructure message section.

- Describe two goals (marking merges and finding merge candidates).

- Add separate sections for each goals and describe different
  solutions for each goal.  For the first goal, specifically
  recomend `git cherry-pick -x` over `MFC <hash>`.

- Add a section on trimmming HEAD/main commit metadata from merges.

bsdjhb committed:
Add an example merge of a subversion commit with a conflict.

bsdjhb committed:
Add an example of a single subversion commit.

bsdjhb commented on MFC.md:
```diff
@@ -7,33 +7,89 @@ is `freebsd` as suggested in other docs.
+
+XXXJHB: I'm not sure what this paragraph means?
 
 There are also advantages to all the commits on stable/X having something to identify them as
```
So this is the paragraph I wasn't sure about as I wasn't sure if you were thinking of using merge commits as an alternative and if so I needed to capture that as a 3rd option.  If that's not the case and this is just saying we want to have some kind of annotation for merges, I think I can just drop this paragraph.

bsdimp replied:
If we mark all the MFCs as MFCs in a unique way, we can have a commit hook for main that rejects them. That's all.

bsdjhb committed:
Further refinements to MFC rationale.

- Add a Goals section stating some goals of merge changes and
  documenting how those goals were implemented in subversion.

- Explicitly talk about pushes being atomic and how that can
  still satisfy the new stability goal for the non-squash
  option.  Also, mention 'git bisect skip' for this option.

- Make trimming metadata an open question rather than a mandate.

  Add various commentary on historical practice and reasons why
  it may or may not make sense to alter metadata in merge commits.
